### PR TITLE
test/add new test for analyzer props file not getting imported

### PR DESCRIPTION
**Pull request by jmarolf**

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.props [hunk 1]
```diff
@@ -10,6 +10,6 @@ Copyright (c) .NET Foundation. All rights reserved.
 ***********************************************************************************************
 -->
 <Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="..\analyzers\build\Microsoft.CodeAnalysis.NetAnalyzers.props"
+  <Import Project="$(MSBuildThisFileDirectory)..\analyzers\build\Microsoft.CodeAnalysis.NetAnalyzers.props"
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.props:13]: Project imports should already be relative to the current file.  Was something failing without this change?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUsePlatformAnalyzers.cs [hunk 2]
```diff
@@ -255,5 +256,52 @@ static void Main()
...
+                    ",
+                }
+            };
+
+            testProject.AdditionalProperties.Add("CodeAnalysisTreatWarningsAsErrors", "false");
+            testProject.AdditionalProperties.Add("TreatWarningsAsErrors", "true");
+            var testAsset = _testAssetsManager
+                .CreateTestProject(testProject, identifier: "analyzerConsoleApp", targetExtension: ".csproj");
+
+            var buildCommand = new GetValuesCommand(
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUsePlatformAnalyzers.cs:296]: A `GetValuesCommand` is for reading a property or item from the build.  It doesn't look like you're doing that here.  Should this be `BuildCommand` instead?

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUsePlatformAnalyzers.cs [hunk 3]
```diff
@@ -255,5 +256,52 @@ static void Main()
...
+                Log,
+                Path.Combine(testAsset.TestRoot, testProject.Name),
+                targetFrameworkNetCore31, "Analyzer")
+            {
+                DependsOnTargets = "Build"
+            };
+            var buildResult = buildCommand.Execute();
+            buildResult.StdErr.Should().Be(string.Empty);
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToUsePlatformAnalyzers.cs:304]: Generally the way we do this is `buildResult.Should().Pass()`.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.props (hunk 1 above)
Comment (jmarolf) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.props:13]: it was not. I verified that this does the right though without this change using -pp. I'll back this out. I was being paranoid.

**Pull request merged**